REGRESSION(r293703):[ BigSur+ iOS ] TestWTF.WTF_URLExtras.URLExtras_ParsingError (API-Test) is a constant failure

cdumez opened the pull request:
#### 555cf8634ec54e5978011e20f665831e8e803607
<pre>
REGRESSION(r293703):[ BigSur+ iOS ] TestWTF.WTF_URLExtras.URLExtras_ParsingError (API-Test) is a constant failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=240049">https://bugs.webkit.org/show_bug.cgi?id=240049</a>

Reviewed by Yusuke Suzuki.

We need to pass the length of the string, which is the number of characters without the null terminator.
However, we were passing utf16.size(), which was one too many since the utf16 array contains the null
terminator at the end.

* Tools/TestWebKitAPI/Tests/WTF/cocoa/URLExtras.mm:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/250240@main">https://commits.webkit.org/250240@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@293761">https://svn.webkit.org/repository/webkit/trunk@293761</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

Constellation commented on Tools/TestWebKitAPI/Tests/WTF/cocoa/URLExtras.mm:
```diff
@@ -234,7 +234,7 @@
     EXPECT_STREQ([[url2 absoluteString] UTF8String], "http://%E2%89%A7%E2%88%AE%EF%B9%A3%D9%A1%DB%B1");
 
     std::array<UChar, 3> utf16 { 0xC2, 0xB6, 0x00 };
```
Let's just drop the last `0x00` and make it `std::array<UChar, 2> utf16`.

Constellation reviewed (approved):
r=me with change.

zdobersek committed:
[RISCV64] Implement MacroAssemblerRISCV64 move-conditionally methods
https://bugs.webkit.org/show_bug.cgi?id=239998

Reviewed by Yusuke Suzuki.

Provide implementations for the variants of the move-conditionally
operation in MacroAssemblerRISCV64. These are true macro operations,
often requiring scratch registers and branches to implement the
behavior since the RISC-V ISA doesn't provide appropriate instructions
out-of-the-box.

Test cases in testmasm are also enabled, including some additional
guards to avoid unused-variable warnings at build-time.

* assembler/MacroAssemblerRISCV64.h:
(JSC::MacroAssemblerRISCV64::moveConditionally32):
(JSC::MacroAssemblerRISCV64::moveConditionally64):
(JSC::MacroAssemblerRISCV64::moveConditionallyFloat):
(JSC::MacroAssemblerRISCV64::moveConditionallyDouble):
(JSC::MacroAssemblerRISCV64::moveConditionallyTest32):
(JSC::MacroAssemblerRISCV64::moveConditionallyTest64):
(JSC::MacroAssemblerRISCV64::moveDoubleConditionally32):
(JSC::MacroAssemblerRISCV64::moveDoubleConditionally64):
(JSC::MacroAssemblerRISCV64::moveDoubleConditionallyFloat):
(JSC::MacroAssemblerRISCV64::moveDoubleConditionallyDouble):
(JSC::MacroAssemblerRISCV64::moveDoubleConditionallyTest32):
(JSC::MacroAssemblerRISCV64::moveDoubleConditionallyTest64):
(JSC::MacroAssemblerRISCV64::branchForMoveConditionally):
* assembler/testmasm.cpp:
(JSC::testProbeModifiesStackPointer):
(JSC::testProbeModifiesStackValues):

Canonical link: https://commits.webkit.org/250238@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@293759 268f45cc-cd09-0410-ab3c-d52691b4dbfc

cdumez committed:
Replace String::remove() by a makeStringByRemoving() free function
https://bugs.webkit.org/show_bug.cgi?id=239995

Reviewed by Yusuke Suzuki.

Replace String::remove() by a makeStringByRemoving() free function. This is a
step towards making String immutable.

* Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm:
(WebKit::lastCNAMEDomain):
* Source/WebKit/UIProcess/Inspector/win/InspectorResourceURLSchemeHandler.cpp:
(WebKit::InspectorResourceURLSchemeHandler::platformStartTask):
* Source/WTF/wtf/URL.cpp:
(WTF::URL::remove):
* Source/WTF/wtf/text/WTFString.cpp:
(WTF::makeStringByRemoving):
(WTF::String::removeInternal): Deleted.
(WTF::String::remove): Deleted.
* Source/WTF/wtf/text/WTFString.h:
* Source/WebCore/Modules/mediastream/RTCDTMFSender.cpp:
(WebCore::RTCDTMFSender::playNextTone):
* Source/WebCore/dom/CharacterData.cpp:
(WebCore::CharacterData::deleteData):
* Source/WebCore/dom/FragmentDirectiveParser.cpp:
(WebCore::FragmentDirectiveParser::parseFragmentDirective):
* Source/WebCore/dom/Range.cpp:
(WebCore::processContentsBetweenOffsets):
* Source/WebCore/editing/CompositeEditCommand.cpp:
(WebCore::CompositeEditCommand::deleteInsignificantText):
* Source/WebCore/inspector/InspectorOverlayLabel.cpp:
(WebCore::InspectorOverlayLabel::draw):
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::InspectorStyleSheet::deleteRule):
* Source/WebCore/platform/graphics/gstreamer/mse/GStreamerMediaDescription.cpp:
(WebCore::GStreamerMediaDescription::extractCodecName):
* Source/WebCore/platform/network/curl/CookieJarDB.cpp:
(WebCore::CookieJarDB::deleteCookie):
* Source/WebCore/platform/text/win/LocaleWin.cpp:
(WebCore::LocaleWin::shortTimeFormat):

Canonical link: https://commits.webkit.org/250239@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@293760 268f45cc-cd09-0410-ab3c-d52691b4dbfc

cdumez committed:
REGRESSION(r293703):[ BigSur+ iOS ] TestWTF.WTF_URLExtras.URLExtras_ParsingError (API-Test) is a constant failure
https://bugs.webkit.org/show_bug.cgi?id=240049

Reviewed by Yusuke Suzuki.

We need to pass the length of the string, which is the number of characters without the null terminator.
However, we were passing utf16.size(), which was one too many since the utf16 array contains the null
terminator at the end.

* Tools/TestWebKitAPI/Tests/WTF/cocoa/URLExtras.mm:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/250240@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@293761 268f45cc-cd09-0410-ab3c-d52691b4dbfc
